### PR TITLE
Change config semantics to args and full key path

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

changes config feature semantics from flags to args

`tctl config set --namespace accounting` 

to 

`tctl config set env.local.namespace accounting`  (or `tctl config set namespace accounting` for short in a new PR)

## Why?

- moving towards allowing to configure the default value for any flag and not just the ones whitelisted by `tctl config set` command   
- consistency with other tools such as `git` and `kubectl` 

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: cli/config.go [hunk 1]
```diff
@@ -155,50 +54,38 @@ func newConfigCommands() []*cli.Command {
...
-			if err := tctlConfig.SetByEnvironment(c, key, val); err != nil {
-				return fmt.Errorf("unable to set property %s: %s", key, err)
-			}
-		} else {
-			return fmt.Errorf("unable to set property %s: invalid key", key)
-		}
-		fmt.Printf("%v: %v\n", color.Magenta(c, "%v", key), val)
+	key := c.Args().Get(0)
+
+	var val string
```
Comment (bergundy) [inline cli/config.go:78]: When is it valid to omit the value?

**Review by bergundy — COMMENTED** [review]
For someone like me who's not familiar with the codebase a PR description would go a long way.
Also it doesn't make sense to me not to have any test coverage for this.

File: cli/config.go (hunk 1 above)
Comment (feedmeapples) [reply cli/config.go:78]: was thinking of booleans such as `tls-disable-host-verification`

**Comment by feedmeapples** [comment]
> For someone like me who's not familiar with the codebase a PR description would go a long way. Also it doesn't make sense to me not to have any test coverage for this.

right, missed adding description after removing the draft tag

File: cli/config.go (hunk 1 above)
Comment (feedmeapples) [reply cli/config.go:78]: so `tctl config set tls-disable-host-verification` would mean it being set as `true`

Comment (bergundy) [reply cli/config.go:78]: Hmm.. without going into too much detail here, how do we deal with booleans and numbers in general?

Comment (feedmeapples) [reply cli/config.go:78]: made them to work with the standard urfave/cli syntax https://github.com/temporalio/tctl/pull/257

**Review by alexshtin — APPROVED** [review]

**Pull request merged**